app/services/price_policies/time_based_price_calculator.rb

```diff
@@ -14,8 +14,8 @@ def initialize(price_policy)
     def calculate(start_at, end_at)
       return if start_at > end_at
       duration_mins = TimeRange.new(start_at, end_at).duration_mins
-      discount = calculate_discount(start_at, end_at)
-      cost_and_subsidy(duration_mins, discount)
+      discount_multiplier = product.try(:schedule_rules) ? calculate_discount(start_at, end_at) : 1
```

Comment:
If we are going to keep this around, I would lean towards using `respond_to?` rather than `try` as it conveys the intention a little better.